config/configmapprovider/file.go

```diff
@@ -16,34 +16,48 @@ package configmapprovider // import "go.opentelemetry.io/collector/config/config
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"io/ioutil"
+	"net/url"
+	"path"
 
 	"gopkg.in/yaml.v2"
 
 	"go.opentelemetry.io/collector/config"
 )
 
-type fileMapProvider struct {
-	fileName string
+const fileSchemeName = "file"
+
+type fileMapProvider struct{}
+
+// NewFile returns a new Provider that reads the configuration from a file.
+//
+// This Provider supports "file" scheme (see https://datatracker.ietf.org/doc/html/rfc8089), and can be called with an
+// uri with or without the scheme name:
+// `/path/to/file` - no scheme, no hostname
+// `file:/path/to/file` - no hostname
```

Comment:
Would this mean that relative paths are no longer supported?

Reply:
They are, I even have a test:

```
func TestFile(t *testing.T) {
	fp := NewFile()
	ret, err := fp.Retrieve(context.Background(), mustParseURL(filepath.Join("testdata", "default-config.yaml")), nil)
	require.NoError(t, err)
	cfg, err := ret.Get(context.Background())
	assert.NoError(t, err)
	expectedMap := config.NewMapFromStringMap(map[string]interface{}{
		"processors::batch":         nil,
		"exporters::otlp::endpoint": "localhost:4317",
	})
	assert.Equal(t, expectedMap, cfg)
	assert.NoError(t, ret.Close(context.Background()))
	assert.NoError(t, fp.Shutdown(context.Background()))
}
```

will update comment to reflect that it is still possible.

Reply:
Done.